src/interop/js-interop/usage.md

```diff
@@ -0,0 +1,454 @@
+  are a factory.
+
+  - **Object literal constructors**. It is useful sometimes to create a JS
+    [object literal] that simply contains a number of properties and their
+    values. In order to do this, you declare a constructor with only named
+    parameters, as you're creating an object that is defined by the names of
+    its properties:
+
+    ```dart
+    import ‘dart:js_interop’;
+
+    extension type Options._(JSObject o) {
+      external Options({int a, int b});
+      external int get a;
+      external int get b;
+    }
+    ```
+
+    A call to `Options(a: 0, b: 1)` will result in creating the JS object
+    `{a: 0, b: 1}`. The object is defined by the invocation, so calling
+    `Options(a: 0)` would result in `{a: 0}`. You can get or set the properties
+    of the resulting object through `external` instance members.
+
+    {{site.alert.warn}}
+    There's a bug that currently requires object literal constructors to have an
+    [`@JS`](#js) annotation on the library, empty or otherwise. See [#54801] for
+    more details.
+    {{site.alert.end}}
+
+- **`static` members**. Like constructors, these members use the name of the
+  extension type to generate the JS code. For example, calling
+  `Time.getTimeDifference(t1, t2)` will generate a JS invocation that looks like
+  `Time.getTimeDifference(t1, t2)`. Similarly, calling `Time.dinnerTime` will
+  result in a JS invocation that looks like `Time.dinnerTime`. Like top-levels,
+  you can declare `static` methods, getters, setters, and fields.
+
+- **Instance members**. Like with other Dart types, these members require an
+  instance in order to be used. These members get, set, or invoke properties on
+  the instance. For example:
+
+  ```dart
+    final time = Time(0, 0);
+    print(time.isDinnerTime()); // false
+    final dinnerTime = Time.dinnerTime;
+    time.hours = dinnerTime.hours;
+    time.minutes = dinnerTime.minutes;
+    print(time.isDinnerTime()); // true
+  ```
+
+  The call to `dinnerTime.hours` gets the value of the `hours` property of
+  `dinnerTime`. Similarly, the call to `time.minutes=` sets the value of the
+  `minutes` property of time. The call to `time.isDinnerTime()` calls the
+  function in the `isDinnerTime` property of `time` and returns the value.
+  Like top-levels and `static` members, you can declare instance methods,
+  getters, setters, and fields.
+
+- **Operators**. There are only two `external` interop operators allowed in
+  interop types: `[]` and `[]=`. These are instance members that match the
+  semantics of JS' [property accessors]. For example, you can declare them like:
+
+  ```dart
+  extension type Array(JSArray<JSNumber> _) implements JSArray<JSNumber> {
+    external JSNumber operator [](int index);
+    external void operator []=(int index, JSNumber value);
+  }
+  ```
+
+  Calling `array[i]` gets the value in the `i`th slot of `array`, and
+  `array[i]=` sets the value in that slot to `i.toJS`. Other JS operators are
```

Comment:
Not sure we need the bit about `i.toJS` anymore.

```suggestion
  `array[i]=` sets the value in that slot. Other JS operators are
```

Reply:
Ah, this was meant to be `array[i] = i.toJS` so that there's a value to refer to.